crates/core/tedge_api/src/alarm.rs

```diff
@@ -21,13 +27,16 @@ pub enum AlarmSeverity {
 }
 
 /// In-memory representation of ThinEdge JSON alarm payload
-#[derive(Debug, Deserialize, Eq, PartialEq)]
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
 pub struct ThinEdgeAlarmData {
     pub text: Option<String>,
 
     #[serde(default)]
     #[serde(with = "time::serde::rfc3339::option")]
     pub time: Option<Timestamp>,
+
+    #[serde(flatten)]
+    pub extras: HashMap<String, Value>,
```

Comment:
I agree, we can rename it as `fragments`. But it will also contain the `child device` info as part of the message. Can this child device info also be called a fragment? I am not sure. So, I kept it in line with the `events`.

Reply:
If the `child-device-id` is stored as an entry in this `extras` map, what's the content of the `source` ? I though it was the id of the child device sending the alarm?

Reply:
I renamed `extras` as `fragments`